panic: (*zed.TypeError) with Go v1.21

I happened to let `brew` on my Macbook update Go to the latest version v1.21. When I later compiled my Zed binaries, one of the mdtests started failing due to what I can repro here:

```
$ go version
go version go1.21.0 darwin/amd64

$ zq -version
Version: v1.9.0-10-g08fc271c

$ echo '0' | zq -z 'yield error(this)' -
panic: (*zed.TypeError) 0xc0000d6ba0

goroutine 1 [running]:
github.com/brimdata/zed.(*Value).Bytes(0xc0005cf050)
	/Users/runner/zed/value.go:152 +0x1f8
github.com/brimdata/zed/zson.(*Formatter).FormatRecord(0xc0004a4990, 0xc0005cf050)
	/Users/runner/zed/zson/formatter.go:75 +0x96
github.com/brimdata/zed/zio/zsonio.(*Writer).Write(0xc0000d69a8, 0xc00041faa8?)
	/Users/runner/zed/zio/zsonio/writer.go:33 +0x25
github.com/brimdata/zed/zbuf.WriteBatch({0xbb59518, 0xc0000d69a8}, {0x28624d0?, 0xc0005cf0b0?})
	/Users/runner/zed/zbuf/batch.go:41 +0x6d
github.com/brimdata/zed/zbuf.CopyPuller({0xbb59518, 0xc0000d69a8}, {0x284d9a0, 0xc0005cf020})
	/Users/runner/zed/zbuf/batch.go:166 +0x76
github.com/brimdata/zed/cli/zq.(*Command).Run(0xc0004fe380, {0xc000112060, 0x2, 0x2})
	/Users/runner/zed/cli/zq/command.go:162 +0x76b
github.com/brimdata/zed/pkg/charm.path.run({0xc00006c238?, 0x1, 0x1}, {0xc000112060?, 0x2, 0x0?})
	/Users/runner/zed/pkg/charm/path.go:11 +0x79
github.com/brimdata/zed/pkg/charm.(*Spec).ExecRoot(0x100874b?, {0xc000112050, 0x3, 0x3})
	/Users/runner/zed/pkg/charm/charm.go:63 +0x3f
main.main()
	/Users/runner/zed/cmd/zq/main.go:11 +0x5b
```

However, if I stay behind on the prior Go release, it runs fine.

```
$ go version
go version go1.20.7 darwin/amd64

$ zq -version
Version: v1.9.0-10-g08fc271c

$ echo '0' | zq -z 'yield error(this)' -
error(0)
```

 I know our `go.mod` specifies `1.20` so this was ultimately my mistake and we've done nothing yet to set incorrect expectations with users. However, I know we tend to try moving up to newer versions of tooling shortly after they're released, so I figured I'd give an early heads-up about this one.

--- a/value.go
+++ b/value.go
@@ -157,7 +157,7 @@ func (v *Value) Bytes() zcode.Bytes {
 // nativeBase is the base address for all native Values, which are encoded as a
 // zcode.Bytes with this base address, a length of zero, and capacity set to the
 // bits of the value's native representation.
-var nativeBase struct{}
+var nativeBase uintptr
 
 func encodeNative(x uint64) zcode.Bytes {
 	var b zcode.Bytes

